lib/web/apiserver.go

```diff
@@ -341,6 +341,63 @@ type APIHandler struct {
 // ConnectionHandler defines a function for serving incoming connections.
 type ConnectionHandler func(ctx context.Context, conn net.Conn) error
 
+func (h *APIHandler) handlePreflight(w http.ResponseWriter, r *http.Request) {
+	raddr, err := utils.ParseAddr(r.Host)
+	if err != nil {
+		return
+	}
+	publicAddr := raddr.Host()
+
+	servers, err := app.Match(r.Context(), h.handler.cfg.AccessPoint, app.MatchPublicAddr(publicAddr))
+	if err != nil {
+		h.handler.log.Error("failed to match application with public addr %s", publicAddr)
+		return
+	}
+
+	if len(servers) == 0 {
+		h.handler.log.Error("failed to match application with public addr %s", publicAddr)
+		return
+	}
+
+	foundApp := servers[0].GetApp()
+	corsPolicy := foundApp.GetCORS()
+	if corsPolicy == nil {
+		return
```

Comment:
Given that we intercept every single cors request, won't this break existing apps without cors specified? 
Should we mark it as breaking change?

Reply:
The wouldn't be a breaking change.
1. Every prelight request is already "broken" so nothing would change
2. its opt in, so only users who _wanted_ CORS to work would have to explicitly set it.